### DISCUSSION TITLE
Multiple cell templates in DataGrid

### DISCUSSION THREAD

**Discussion by agnauck** [AvaloniaUI/Avalonia#10710] [Q&A] [answered]
The DataGrid seems to support only 1 template per cell. I often use multiple data temples based on the bound item like described below.

What would be the best approach to do this with the DataGrid?

```
<DataTemplates>
  <DataTemplate DataType="Cat">
     <TextBlock Text="Cat">
  </DataTemplate>
  <DataTemplate DataType="Dog">
     <Button Content="Dog">
  </DataTemplate>
</DataTemplates>
```

Thanks,
Alex

**Comment by timunie** [comment, marked as answer]
Hey @agnauck ,

I see different options here: 
- Add a wraprring `ContentControl` to your `ItemTemplate`, which has support for `DataTemplates` as written above
- Use `FuncDataTemplate` (see https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/DataTemplates/FuncDataTemplateSample) 
- Implement `IDataTemplate` (see: https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/DataTemplates/IDataTemplateSample)

**Reply by agnauck** [reply]
Thanks, I went which option 3 that worked fine for me 👍
